### ISSUE TITLE
✅ 🐞 Bug report: Icons missing from form submission message

### ISSUE THREAD

**Issue by jamieburchell** [elementor/elementor#19557]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

With the Inline Font Icons experiment enabled, icons are missing from confirmation messages:

![image](https://user-images.githubusercontent.com/3270426/185894742-b91741ab-eb5a-41ff-b820-df49a9ec38d7.png)

### Steps to reproduce

Enable the Inline Font Icons experiment, send a message through the form widget

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MariaDB Server v10.5.16
	PHP Version: 8.0.22
	PHP Memory Limit: 256M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 50M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: There are some writing permissions issues with the following directories/files:
		 - WordPress root directory
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.0.1
	Site URL: https://example.com/wp
	Home URL: https://example.com
	WP Multisite: No
	Max Upload Size: 40 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: en-GB
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.6.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_GB
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36

== Active Plugins ==
	Code Snippets
		Version: 3.2.0
		Author: Code Snippets Pro

	Dynamic.ooo - Dynamic Content for Elementor
		Version: 2.7.5
		Author: Dynamic.ooo

	Elementor
		Version: 3.7.2
		Author: Elementor.com

	Elementor Pro
		Version: 3.7.3
		Author: Elementor.com

	Pods - Custom Content Types and Fields
		Version: 2.9.3
		Author: Pods Framework Team

	Yoast SEO
		Version: 19.5.1
		Author: Team Yoast


== Must-Use Plugins ==
	Bedrock Autoloader
		Version: 1.0.3
		Author: Roots

	Disable status tests
		Version: 1.0.0
		Author: Roots

	Register Theme Directory
		Version: 1.0.0
		Author: Roots


== Elementor Experiments ==
	Optimised DOM Output: Inactive by default
	Improved asset loading: Inactive by default
	Improved CSS Loading: Active
	Inline Font Icons: Active
	Accessibility improvements: Inactive by default
	Additional Custom Breakpoints: Inactive by default
	Import Export Template Kit: Active by default
	Hide native WordPress widgets from search results: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Inactive by default
	Landing pages: Active by default
	Color Sampler: Active by default
	Favorite Widgets: Active by default
	Admin Top Bar: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default


== Log ==
	
JS: showing 19 of 19JS: 2022-08-05 10:09:22 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view1706\") has already been destroyed and cannot be used. 
JS: 2022-08-05 11:31:26 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view458\") has already been destroyed and cannot be used. 
JS: 2022-08-05 11:31:30 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view2638\") has already been destroyed and cannot be used. 
JS: 2022-08-09 09:37:16 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view1673\") has already been destroyed and cannot be used. 
JS: 2022-08-09 09:37:16 [error X 4][../plugins/elementor/assets/js/frontend-modules.min.js?ver=3.6.8:2:4766] Cannot read properties of undefined (reading \'attributes\') 
JS: 2022-08-09 09:38:37 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view2050\") has already been destroyed and cannot be used. 
JS: 2022-08-09 09:59:48 [error X 2][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view135\") has already been destroyed and cannot be used. 
JS: 2022-08-09 10:16:09 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view56376\") has already been destroyed and cannot be used. 
JS: 2022-08-09 10:16:09 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view56371\") has already been destroyed and cannot be used. 
JS: 2022-08-09 10:18:41 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view198\") has already been destroyed and cannot be used. 
JS: 2022-08-09 10:18:46 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view2254\") has already been destroyed and cannot be used. 
JS: 2022-08-09 13:26:06 [error X 2][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view212\") has already been destroyed and cannot be used. 
JS: 2022-08-09 13:27:35 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view1685\") has already been destroyed and cannot be used. 
JS: 2022-08-09 13:27:54 [error X 1][../plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view8434\") has already been destroyed and cannot be used. 
JS: 2022-08-09 13:37:25 [error X 7][../plugins/elementor/assets/js/editor.min.js?ver=3.6.8:3:900529] Cannot read properties of null (reading \'getBoundingClientRect\') 
JS: 2022-08-10 23:28:08 [error X 1][https://example.com/wp/wp-includes/js/jquery/jquery.min.js?ver=3.6.0:2:13633] Syntax error, unrecognized expression: #elementor-control-left: auto; right: 0;-c19854 
JS: 2022-08-10 23:34:08 [error X 1][https://example.com/wp/wp-includes/js/jquery/jquery.min.js?ver=3.6.0:2:13633] Syntax error, unrecognized expression: #elementor-control-left: 0%;-c1799 
JS: 2022-08-10 23:34:35 [error X 1][https://example.com/wp/wp-includes/js/jquery/jquery.min.js?ver=3.6.0:2:13633] Syntax error, unrecognized expression: #elementor-control-left: auto; right: 0;-c1799 
JS: 2022-08-15 13:22:59 [error X 11][../plugins/elementor/assets/js/common-modules.min.js?ver=3.7.0:2:51471] Model.isValidChild() should be implemented, please provide \'isValidChild\' functionality. 

PHP: showing 5 of 5PHP: 2022-08-09 13:25:45 [notice X 4][../plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-08-10 23:24:56 [warning X 118][../plugins/elementor/core/common/modules/ajax/module.php::171] Undefined array key "data" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-08-15 11:59:28 [warning X 62][../plugins/elementor-pro/modules/dynamic-tags/pods/tags/pods-text.php::20] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-08-15 15:21:52 [error X 2][../plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::294] Uncaught Error: Call to a member function update_meta() on null in ../plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:294
Stack trace:
#0 ../plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(167): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(5, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 ../plugins/elementor/core/common/modules/ajax/module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 ../wp-includes/class-wp-hook.php(307): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 ../wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters('', Array)
#5 ../wp-includes/plugin.php(476): WP_Hook->do_action(Array)
#6 ../wp-admin/admin-ajax.php(187): do_action('wp_ajax_element...')
#7 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-08-17 14:54:14 [error X 2][../plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::294] Uncaught Error: Call to a member function update_meta() on null in ../plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:294
Stack trace:
#0 ../plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(167): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(5, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 ../plugins/elementor/core/common/modules/ajax/module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 ../wp-includes/class-wp-hook.php(307): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 ../wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters('', Array)
#5 ../wp-includes/plugin.php(476): WP_Hook->do_action(Array)
#6 ../wp-admin/admin-ajax.php(187): do_action('wp_ajax_element...')
#7 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Log: showing 18 of 182022-08-10 22:58:08 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.8',
  'to' => '3.7.0',
)]
2022-08-10 23:04:02 [info] elementor::elementor_updater Started 
2022-08-10 23:04:02 [info] Elementor/Upgrades - _on_each_version Start  
2022-08-10 23:04:02 [info] Elementor/Upgrades - _on_each_version Finished 
2022-08-10 23:04:02 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.6.8',
  'to' => '3.7.0',
)]
2022-08-10 23:19:19 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.7.2',
  'to' => '2.7.4',
)]
2022-08-17 14:51:55 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.0',
  'to' => '3.7.1',
)]
2022-08-17 14:52:12 [info] Update DB has been started [array (
  'plugin' => 'Elementor',
  'from' => '3.7.0',
  'to' => '3.7.1',
)]
2022-08-17 14:52:12 [info] elementor::elementor_updater Started 
2022-08-17 14:52:12 [info] Elementor/Upgrades - _on_each_version Start  
2022-08-17 14:52:12 [info] Elementor/Upgrades - _on_each_version Finished 
2022-08-17 14:52:12 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.0',
  'to' => '3.7.1',
)]
2022-08-22 09:28:42 [info] Update DB has been started [array (
  'plugin' => 'Elementor',
  'from' => '3.7.1',
  'to' => '3.7.2',
)]
2022-08-22 09:28:42 [info] elementor::elementor_updater Started 
2022-08-22 09:28:42 [info] Elementor/Upgrades - _on_each_version Start  
2022-08-22 09:28:42 [info] Elementor/Upgrades - _on_each_version Finished 
2022-08-22 09:28:42 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.1',
  'to' => '3.7.2',
)]
2022-08-22 09:35:36 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Dynamic.ooo - Dynamic Content for Elementor',
  'from' => '2.7.4',
  'to' => '2.7.5',
)]



== Elementor - Compatibility Tag ==
	
	Dynamic.ooo - Dynamic Content for Elementor: Compatibility not specified
	Elementor Pro: Compatibility not specified

== Elementor Pro - Compatibility Tag ==

````

</details>

**Comment by jamieburchell** [comment]
@rami-elementor Not sure if it really warrants a completely new issue, but the icons are also missing from the next and previous links on the Post Navigation widget.

**Comment by rami-elementor** [comment]
@jamieburchell Please create a new issues for the next and previous links.

P.S. To fix the issue, you need to deactivate the "**Inline Font Icons**" experiment. It's not a stable experiment, it's an alpha stage experiment. There are several known bugs that are currently are being fixed.

**Comment by jamieburchell** [comment]
@rami-elementor Can you point me to the list of known bugs so I don't waste time reporting known issues?

**Comment by rami-elementor** [comment]
We have an internal tools to monitor those bugs.

But as I mentioned, "alpha" experiments are not "stable" experiments, you should not activate them. There are only two alpha experiments, try to avoid them for now.

**Comment by jamieburchell** [comment]
This experiment has since transitioned to beta status and this issue is still present

**Comment by rloes** [comment]
+1

**Comment by rloes** [comment]
For the meantime, this snippet added to your function.php should do the trick. It loads the elementor-icons font only on pages where a elementor form is rendered. The elementor-icons font isnt loaded if the icon experiment is active.

```php
add_action('elementor-pro/forms/pre_render', function($instance, $form){
	wp_enqueue_style( 'elementor-icons' );
}, 10, 2);
```

**Comment by jamieburchell** [comment]
This experiment becomes active by default in the upcoming version; is this fixed?

**Comment by inamullah0346** [comment]
the problem is still not solve
![Uploading Capture.PNG…]()
d why??? we have elementor pro 19.2 but still showing the problem

**Comment by j-menter** [comment]
I also still have this issue. i solved it with a little js script i placed inside of a html widget:

`
<script>
document.addEventListener('DOMContentLoaded', function() {
    var observer = new MutationObserver(function(mutations) {
        mutations.forEach(function(mutation) {
            mutation.addedNodes.forEach(function(node) {
                if (node.nodeType === 1 && node.classList.contains('elementor-message-success')) {
                    // Überschreibe den Inhalt der Erfolgsmeldung
                    node.innerHTML = 'Deine Anfrage wurde erfolgreich versendet';
                    // Füge eine spezielle Klasse hinzu, um ::before zu verstecken
                    node.classList.add('hide-before');
                }
            });
        });
    });

    var config = { childList: true, subtree: true };
    observer.observe(document.body, config);

    // CSS um ::before bei der spezifischen Klasse zu verstecken
    var styleSheet = document.createElement('style');
    styleSheet.type = 'text/css';
    styleSheet.innerText = '.hide-before::before { display: none !important; }';
    document.head.appendChild(styleSheet);
});

</script>`

Elementor - 3.21.5
Elementor Pro - 3.21.2

**Comment by rloes** [comment]
@j-menter If you want the icons to display, you can add this php snippet to functions.php or code snippets plugin.

This way you dont have to hide the wrongly displayed icon, dynamically using JS:

```php
add_action('elementor-pro/forms/pre_render', function($instance, $form){
	wp_enqueue_style( 'elementor-icons' );
}, 10, 2);
```

**Comment by j-menter** [comment]
> @j-menter If you want the icons to display, you can add this php snippet to functions.php or code snippets plugin.
> 
> This way you dont have to hide the wrongly displayed icon, dynamically using JS:
> 
> ```
> add_action('elementor-pro/forms/pre_render', function($instance, $form){
> 	wp_enqueue_style( 'elementor-icons' );
> }, 10, 2);
> ```

I know. But i think its bad practice to place anything in function.php. Exprecually for non-child themes and when i want to make any updates without getting into the same struggles again. For that i like the js solution more.

**Comment by incircolo** [comment]
> @j-menter If you want the icons to display, you can add this php snippet to functions.php or code snippets plugin.
> 
> This way you dont have to hide the wrongly displayed icon, dynamically using JS:
> 
> ```
> add_action('elementor-pro/forms/pre_render', function($instance, $form){
> 	wp_enqueue_style( 'elementor-icons' );
> }, 10, 2);
> ```

Hi
Where do you find this hook "elementor-pro/forms/pre_render" ?
I'm curios to know why this is "elementor-pro" and not "elementor_pro". I don't find it in elementor developer docs.

**Comment by TOTLD** [comment]
Issue is persistent. I'm facing it in  Elementor v3.22.1 + Elementor Pro v3.22.0 using Hello.

Pretty wild that the inline icons feature has gone from alpha to beta to stable over almost 2 years without this being addressed. Particularly when flagged as a known bug by developers here almost 2 years ago too.

**Comment by rloes** [comment]
@incircolo I found it in the elementor pro code. But you right it isn't officially documented and it should probably be elementor_pro. So it could be changed in the future. Maybe when they document it officially. However it works at least until 3.21

**Comment by dkiryakov** [comment]
Also having the issue here. Disabling "Inline Font Icons" feature fixed it, but impacts performance.

**Comment by karlaoviedo21** [comment]
> @j-menter If you want the icons to display, you can add this php snippet to functions.php or code snippets plugin.
> 
> This way you dont have to hide the wrongly displayed icon, dynamically using JS:
> 
> ```
> add_action('elementor-pro/forms/pre_render', function($instance, $form){
> 	wp_enqueue_style( 'elementor-icons' );
> }, 10, 2);
> ```

thank you!

**Comment by jamieburchell** [comment]
This appears to be resolved in 3.23.0, but unfortunately [this experiment still has issues](https://github.com/elementor/elementor/issues/28064).

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v2.23.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
